integration/tests/tracer/test_tracer_historical_methods.py

```diff
@@ -0,0 +1,293 @@
+import random
+import typing as tp
+import allure
+import pytest
+from integration.tests.basic.helpers.basic import BaseMixin
+from utils.helpers import wait_condition
+
+CONTRACT_CODE = '6060604052600080fd00a165627a7a72305820e75cae05548a56ec53108e39a532f0644e4b92aa900cc9f2cf98b7ab044539380029'
+DEPLOY_CODE = '60606040523415600e57600080fd5b603580601b6000396000f300' + CONTRACT_CODE
+
+
+@allure.story("Tracer API RPC calls historical methods check")
+class TestTracerRpcCalls(BaseMixin):
+    _contract: tp.Optional[tp.Any] = None
+
+    @pytest.fixture
+    def storage_contract(self) -> tp.Any:
+        if not TestTracerRpcCalls._contract:
+            contract, _ = self.web3_client.deploy_and_get_contract(
+                "StorageSoliditySource.sol",
+                "0.8.8",
+                self.sender_account,
+                contract_name="Storage",
+                constructor_args=[],
+            )
+            TestTracerRpcCalls._contract = contract
+        yield TestTracerRpcCalls._contract
+        self.store_value(0, TestTracerRpcCalls._contract)
+
+    def store_value(self, value, storage_contract):
+        nonce = self.web3_client.eth.get_transaction_count(
+            self.sender_account.address
+        )
+        instruction_tx = storage_contract.functions.store(value).build_transaction(
+            {
+                "nonce": nonce,
+                "gasPrice": self.web3_client.gas_price(),
+            }
+        )
+        receipt = self.web3_client.send_transaction(
+            self.sender_account, instruction_tx)
+        assert receipt["status"] == 1
+        return instruction_tx, receipt
+
+    def retrieve_value(self, storage_contract):
+        nonce = self.web3_client.eth.get_transaction_count(
+            self.sender_account.address
+        )
+        instruction_tx = storage_contract.functions.retrieve().build_transaction(
+            {
+                "nonce": nonce,
+                "gasPrice": self.web3_client.gas_price(),
+            }
+        )
+        receipt = self.web3_client.send_transaction(
+            self.sender_account, instruction_tx)
+
+        assert receipt["status"] == 1
+        return instruction_tx, receipt
+
+    def make_tx_object(self, sender=None, receiver=None, tx=None) -> tp.Dict:
+        if sender is None:
+            sender = self.sender_account.address
+
+        tx_call_obj = {
+            "from": sender,
+            "to": receiver,
+            "value": hex(tx["value"]),
+            "gas": hex(tx["gas"]),
+            "gasPrice": hex(tx["gasPrice"]),
+            "data": tx["data"],
+        }
+
+        return tx_call_obj
+
+    def call_storage(self, storage_contract, storage_value, request_type):
+        request_value = None
+        _, _ = self.store_value(storage_value, storage_contract)
```

Comment:
It can be just self.store_value(storage_value, storage_contract) :)

Reply:
done